plasmapy/dispersion/numerical/hirose_.py

```diff
@@ -0,0 +1,316 @@
+        except ChargeError:
+            ion.ionize(n=1, inplace=True)
+            z_mean = abs(ion.charge_number)
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # invalidate T_i argument
+    if "T_i" in kwargs["kwargs"]:
+        raise TypeError(
+            "Got unexpected keyword 'T_i', dispersion relation assumes T_i = 0."
+        )
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must a single value and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if k.ndim not in (0, 1):
+        raise ValueError(
+            f"Argument 'k' needs to be a single valued or 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be 0 or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if theta.ndim not in (0, 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {theta.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = np.array([k.value]) * u.rad / u.m
+
+    # Calc needed plasma parameters
+    n_e = z_mean * n_i
+    v_A = Alfven_speed(B, n_i, ion=ion, z_mean=z_mean).value
+    omega_ci = gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean).value
+    omega_pi = plasma_frequency(n=n_i, particle=ion, z_mean=z_mean).value
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=PhysicsWarning)
+        c_s = ion_sound_speed(
+            T_e=T_e,
+            T_i=0 * u.K,
+            ion=ion,
+            n_e=n_e,
+            gamma_e=gamma_e,
+            gamma_i=gamma_i,
+            z_mean=z_mean,
+        ).value
+
+    thetav, kv = np.meshgrid(theta.value, k.value)
+
+    # Parameter kz
+    kz = np.cos(thetav) * kv
+
+    # Define helpful parameters
+    A = (kz * v_A) ** 2
+    B = (kv * c_s) ** 2
+    C = (kv * v_A) ** 2
+    D = ((kv * c_si_unitless) / omega_pi) ** 2
+
+    # Polynomial coefficients: c3*x^6 + c2*x^4 + c1*x^2 + c0
+    c3 = np.ones_like(A)
+    c2 = -A * (1 + D) - B - C
+    c1 = A * (2 * B + C + B * D)
```

Comment:
```suggestion
    c1 = A * (B * (2 + D) + C)
```

[optional]  This just seems more readable to me.